__tests__/components/editor/Editor.test.js

```diff
@@ -1,55 +1,138 @@
-import SaveAndPublishButton from 'components/editor/SaveAndPublishButton'
-import { Prompt } from 'react-router'
+import { MemoryRouter } from 'react-router-dom'
+
 
 const props = {
   location: { state: { resourceTemplateId: 'resourceTemplate:bf:Note' } },
   userWantsToSave: jest.fn(),
   saveError: 'oops',
 }
 
+
+const createInitialState = (options = {}) => {
+  const state = {
+    selectorReducer: {
+      appVersion: {
+        version: '9.3',
+      },
+      resource: {
+        'resourceTemplate:bf2:WorkTitle': {
+          'http://id.loc.gov/ontologies/bibframe/mainTitle': {
+            items: {},
+          },
+        },
+      },
+      entities: {
+        languages: {
+          options: [{
+            id: 'en',
+            label: 'English',
+          }],
+        },
+        resourceTemplates: {
+          'resourceTemplate:bf2:WorkTitle': {
+            id: 'resourceTemplate:bf2:WorkTitle',
+            resourceLabel: 'Work Title',
+            resourceURI: 'http://id.loc.gov/ontologies/bibframe/Title',
+            propertyTemplates: [
+              {
+                propertyURI: 'http://id.loc.gov/ontologies/bibframe/mainTitle',
+                propertyLabel: 'Preferred Title for Work',
+                remark: 'http://access.rdatoolkit.org/rdachp6_rda6-2036.html',
+                mandatory: 'false',
+                repeatable: 'true',
+                type: 'literal',
+                resourceTemplates: [],
+                valueConstraint: {
+                  valueTemplateRefs: [],
+                  useValuesFrom: [],
+                  valueDataType: {},
+                  defaults: [],
+                },
+              },
+            ],
+          },
+        },
+      },
+      editor: {
+        copyToNewMessage: {
+          timestamp: Date.now(),
+          oldUri: 'https://sinopia.io/pcc/1345',
+        },
+        resourceValidationErrors: {},
+        saveResourceError: 'Error 999',
+        rdfPreview: {
+          show: true,
+        },
+        groupChoice: {
+          show: false,
+        },
+        errors: [],
+      },
+    },
+  }
+
+  if (options.loggedIn) {
+    state.authenticate = {
+      authenticationState: {
+        currentSession: {
+          dummy: 'blah',
+        },
+      },
+    }
+  }
+
+
+  return state
+}
+
 // See https://github.com/nodesecurity/eslint-plugin-security/issues/26
 /* eslint security/detect-non-literal-fs-filename: 'off' */
 
 describe('<Editor />', () => {
   describe('any user', () => {
-    const wrapper = shallow(<Editor.WrappedComponent {...props}/>)
+    const store = createReduxStore(createInitialState())
 
-    it('has div with id "editor"', () => {
-      expect(wrapper.find('div#editor').length).toBe(1)
-    })
     it('renders <ResourceTemplate /> component', () => {
-      expect(wrapper.find(ResourceTemplate).length).toBe(1)
+      const { queryByText } = renderWithRedux(
```

Comment:
Courtesy of @jgreben, there is now a renderWithReduxAndRouter method: https://github.com/LD4P/sinopia_editor/blob/master/__tests__/test_utilities/testUtils.js#L17